js/bybit.js

```diff
@@ -2458,284 +2539,317 @@ module.exports = class bybit extends Exchange {
+            return await this.createSpotOrder (symbol, type, side, amount, price, params);
+        } else if (isUsdcSettled) {
+            return await this.createUsdcOrder (symbol, type, side, amount, price, params);
+        } else {
+            return await this.createContractOrder (symbol, type, side, amount, price, params);
+        }
+    }
+
+    async createSpotOrder (symbol, type, side, amount, price = undefined, params = {}) {
         await this.loadMarkets ();
         const market = this.market (symbol);
+        if (type === 'market' && side === 'buy') {
+            // for market buy it requires the amount of quote currency to spend
+            if (this.options['createMarketBuyOrderRequiresPrice']) {
+                const cost = this.safeNumber (params, 'cost');
+                params = this.omit (params, 'cost');
+                if (price === undefined && cost === undefined) {
+                    throw new InvalidOrder (this.id + " createOrder() requires the price argument with market buy orders to calculate total order cost (amount to spend), where cost = amount * price. Supply a price argument to createOrder() call if you want the cost to be calculated for you from price and amount, or, alternatively, add .options['createMarketBuyOrderRequiresPrice'] = false to supply the cost in the amount argument (the exchange-specific behaviour)");
+                } else {
+                    amount = (cost !== undefined) ? cost : amount * price;
+                }
+            }
+        }
         const request = {
             'symbol': market['id'],
-            // 'order_link_id': 'string', // one of order_id, stop_order_id or order_link_id is required
-            // regular orders ---------------------------------------------
-            // 'order_id': id, // one of order_id or order_link_id is required for regular orders
-            // conditional orders ---------------------------------------------
-            // 'stop_order_id': id, // one of stop_order_id or order_link_id is required for conditional orders
+            'side': this.capitalize (side),
+            'type': type.toUpperCase (), // limit, market or limit_maker
+            'timeInForce': 'GTC', // FOK, IOC
+            'qty': amount,
         };
-        let method = undefined;
-        if (market['swap']) {
-            if (market['linear']) {
-                method = 'privateLinearGetOrderSearch';
-            } else if (market['inverse']) {
-                method = 'v2PrivateGetOrder';
+        if (type === 'limit' || type === 'limit_maker') {
+            if (price === undefined) {
+                throw new InvalidOrder (this.id + ' createOrder requires a price argument for a ' + type + ' order');
             }
-        } else if (market['future']) {
-            method = 'futuresPrivateGetOrder';
+            request['price'] = price;
         }
-        const stopOrderId = this.safeString (params, 'stop_order_id');
```

Comment:
We have lost some of the stop-order functionality here. Let's add the support for stop orders for order-methods like `fetch(*)Order(s)`  and `cancelOrder(s)` to recognize the `{'stop': true}` boolean flag and/or a `stop_order_id` in params.

Reply:
@kroitor we did not lose that functionality I just changed from checking for 'stop_order_id' (I don't think this is very friendly and it is not a unified way to specify that we want a stop order) to this: https://github.com/ccxt/ccxt/pull/13263/files#diff-cb8b530c783df8bccc6e0df910e7433b34c9c9c3cea8904fcf2c59f1e60aeebeR3347, checking for a type in params.

 I'll add the 'stop' then, but do you think that we should keep the 'stop_order_id' ?

Reply:
> I'll add the 'stop' then, but do you think that we should keep the 'stop_order_id' ?

I would keep it, cause if the user specifies it, that's a direct indication of his intent to use stop orders.

Reply:
@kroitor I understand but for instance in methods that require an 'id' can't it be a bit confusing? because the user needs to pass an id to the method (even if is None/Null), what if he passes 2 different and valid ids?

Reply:
On the other hand, let's say `cancelAllOrders` does not require any id so would not be correct to use 'stop_order_id' to indicate "stop orders"

Reply:
> On the other hand, let's say `cancelAllOrders` does not require any id so would not be correct to use 'stop_order_id' to indicate "stop orders"

No-no, i meant stop_order_id is only applicable to single-order methods like fetchOrder, cancelOrder

Reply:
@kroitor yes, I meant that is not a "universal indicator" of stop orders because can't be always used but I'll add it no worries

Reply:
@kroitor  Done 🙂